### PR TITLE
Extended getName to support starfall and expadv

### PR DESCRIPTION
**Pull request by thegrb93**

### CODE REVIEW

**Comment by Divran** [comment]
If I remember correctly, according to the coding style created by TomyLobo, you shouldn't have brackets around if statements.

Also, will this work with the other starfall repo? Just make sure it at least doesn't cause Lua errors with their version and that's fine in my opinion.

**Comment by AbigailBuccaneer** [comment]
I'd rather not have class-specific hacks for this. Something along the lines of `return this.GetName and this:GetName() or ""` (or another well-defined method name) would be nicer. We shouldn't reference external addons in our codebase.

**Comment by Divran** [comment]
`GetName` is already a function in GLua which does something else

**Comment by thegrb93** [comment]
Lets not over-complicate it. These chips rely on wiremod and they all are similar to e2 so It's not like I'm adding random entities to it.

**Comment by Divran** [comment]
(psst if Abigail doesn't agree to this, you can always replace the function by overriding its entry in this table https://github.com/wiremod/wire/blob/master/lua/entities/gmod_wire_expression2/core/init.lua#L162)

**Comment by thegrb93** [comment]
Yeah could do a e2 extension or something, but that'd be going way out of the way for such a simple feature.

File: lua/entities/gmod_wire_expression2/core/selfaware.lua [hunk 1]
```diff
@@ -108,8 +108,14 @@ end
 
 -- Get the name of another E2
 e2function string entity:getName()
-	if (!IsValid(this) or this:GetClass() != "gmod_wire_expression2") then return "" end
-	return this.name or ""
+	if (IsValid(this)) then
+		local class = this:GetClass()
+		if (class == "gmod_wire_expression2" or class == "starfall_processor") then
```
Comment (TomyLobo) [inline lua/entities/gmod_wire_expression2/core/selfaware.lua]: Specific entity classes should not be used here. instead, design an interface that entity classes can implement.
GetGateName sounds good.

**Comment by TomyLobo** [comment]
@Divran Please do not encourage people to break E2. It's bad enough as it is, it doesn't need encouragement.

**Comment by thegrb93** [comment]
That should do it.

**Comment by Divran** [comment]
Seems like the best solution to me 👍

File: lua/entities/gmod_wire_expression2/core/selfaware.lua [hunk 2]
```diff
@@ -108,8 +108,9 @@ end
 
 -- Get the name of another E2
 e2function string entity:getName()
-	if (!IsValid(this) or this:GetClass() != "gmod_wire_expression2") then return "" end
-	return this.name or ""
+	if (IsValid(this) and this.GetGateName) then
+		return this:GetGateName()
+	end
```
Comment (TomyLobo) [inline lua/entities/gmod_wire_expression2/core/selfaware.lua]: ok now you return nil if that if doesn't match.

Comment (Divran) [inline lua/entities/gmod_wire_expression2/core/selfaware.lua]: fixit

Comment (TomyLobo) [inline lua/entities/gmod_wire_expression2/core/selfaware.lua]: the `or ""` should probably be there as well

**Comment by thegrb93** [comment]
Crap

File: lua/entities/gmod_wire_expression2/core/selfaware.lua (hunk 2 above)
Comment (TomyLobo) [inline lua/entities/gmod_wire_expression2/core/selfaware.lua]: the `or ""` is still missing

**Comment by thegrb93** [comment]
It's on line 114, git just isnt updating for w/e reason.

File: lua/entities/gmod_wire_expression2/core/selfaware.lua (hunk 2 above)
Comment (thegrb93) [inline lua/entities/gmod_wire_expression2/core/selfaware.lua]: Line 114

Comment (thegrb93) [inline lua/entities/gmod_wire_expression2/core/selfaware.lua]: Ohhh I know what you mean now. Isn't it save to assume name isn't nil? I guess I'll add it anyway.

File: lua/entities/gmod_wire_expression2/core/selfaware.lua [hunk 3]
```diff
@@ -108,8 +108,10 @@ end
 
 -- Get the name of another E2
 e2function string entity:getName()
-	if (!IsValid(this) or this:GetClass() != "gmod_wire_expression2") then return "" end
-	return this.name or ""
+	if (IsValid(this) and this.GetGateName) then
```
Comment (AbigailBuccaneer) [inline lua/entities/gmod_wire_expression2/core/selfaware.lua]: Minor code style issue brought up by Divran before - unnecessary parentheses around `if` condition.

Comment (thegrb93) [inline lua/entities/gmod_wire_expression2/core/selfaware.lua]: There

**Pull request merged**